### DISCUSSION TITLE
Support templates, compiled by `precompileTemplate`

### DISCUSSION THREAD

**Discussion by lifeart** [typed-ember/glint#526] [Q&A]
Hola, amigos! :))

As I understand, notation for templates like:

```ts
import { hbs } from '@glimmerx/component'
export default hbs(`<div></div>`);
```

is supported by glint for glimmerx case, but, not this notation:

```ts
import { precompileTemplate } from '@ember/template-compilation';
export default precompileTemplate(`<div></div>`);
```

Is not really `low-level`, and it will be great to have it supported by glint by default. 
I could take a look into if smb point me where to start digging.

Trying to get it working for [this](https://github.com/lifeart/demo-ember-vite/blob/master/src/components/HelloWorld/index.ts) case

**Comment by dfreeman** [comment]
> supported by glint for glimmerx case

Glint supports tagged strings for template literals in GlimmerX, but not arbitrary function calls. We've actually dropped our tagged-string support for `ember-template-imports`, and in the future _may_ eventually do so for GlimmerX as well if it also moves toward `<template>`.

> ```hbs
> export default precompileTemplate(`<div></div>`);
> ```

The issue in part is that there's not enough type information available for a given arbitrary `precompileTemplate` call to know what is or isn't legal in the template, i.e. answering questions like "What does `{{this}}` refer to?" whose answers could vary depending on what is ultimately done with the template value in question.

> Is not really low-level, and it will be great to have it supported by glint by default.

The [RFC that introduced `precompileTemplate`](https://github.com/emberjs/rfcs/blob/master/text/0496-handlebars-strict-mode.md#low-level-apis) defined it verbatim as a "low-level API" 😉 

We had a similar conversation about `setComponentTemplate` in Discord last summer (the earlier bits are spread a bit back, but [this message](https://discord.com/channels/480462759797063690/987100636631924746/989149167723114566) is one possible spot to pick up from).
